src/test/java/BaseTest.java

```diff
@@ -4,55 +4,94 @@
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
 import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
 import org.testng.annotations.Parameters;
 
+import java.net.MalformedURLException;
+import java.net.URI;
 import java.time.Duration;
 
 
 public class BaseTest {
-    static WebDriver driver;
-    public String url = null;
+    WebDriver driver;
+    public String url = "https://bbb.testpro.io/";
+    WebDriverWait wait;
 
 
-    @BeforeSuite
-    static void setupClass() {
-        WebDriverManager.chromedriver().setup();
-    }
+//    @BeforeSuite
+//    static void setupClass() {
+//        WebDriverManager.chromedriver().setup();
+//    }
 
     @BeforeMethod
-    @Parameters({"baseURL"})
-    public void setUpBrowser(String baseURL) {
-       url = baseURL;
-        ChromeOptions options = new ChromeOptions();
-        options.addArguments("--disable-notifications");
-        driver = new ChromeDriver(options);
+
+    public void setUpBrowser() throws MalformedURLException {
+//        String url = "https://bbb.testpro.io/";
+//        ChromeOptions options = new ChromeOptions();
+//        options.addArguments("--disable-notifications");
+//        driver = new ChromeDriver(options);
+        driver = pickBrowser (System.getProperty("browser"));
         driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        wait = new WebDriverWait(driver,Duration.ofSeconds(4));
         driver.manage().window().maximize();
         driver.get(url);
     }
 
+    public WebDriver pickBrowser(String browser) throws MalformedURLException{
+        DesiredCapabilities caps = new DesiredCapabilities();
+        String gridUrl = "http://192.168.5.21:4444";
+
+        switch (browser) {
+            case "firefox":
+                WebDriverManager.firefoxdriver().setup();
+                return driver = new FirefoxDriver();
```

Comment:
_You can run Chrome browser  or Firefox using your local Grid:_


**Step 1** Add the **selenium-server-4.6.0.jar** file to the project
Download Remote Selenium WebDriver (Grid)  https://www.selenium.dev/downloads/
![image](https://user-images.githubusercontent.com/104761482/201593677-9db9c5fe-2065-43e1-840b-2e151e2f988e.png)



**Step 2** Add the geckodriver.exe file to the project
Download GeckoDriver  https://github.com/mozilla/geckodriver/releases
![image](https://user-images.githubusercontent.com/104761482/201594441-27ed0ad0-9b71-49ca-9459-9c6c652b7209.png)

![image](https://user-images.githubusercontent.com/104761482/201594603-84a99702-5809-4048-aacc-ec0911433d07.png)

**Step 3**  Add "geckodriver.exe" (Windows) or geckodriver (macOS)
For example
`System.setProperty("webdriver.gecko.driver","geckodriver.exe");`
![image](https://user-images.githubusercontent.com/104761482/201294239-cb6ccd25-6531-405c-ac9b-bb2fc09cbdda.png)

```
public WebDriver pickBrowser(String browser) throws MalformedURLException {
    DesiredCapabilities capabilities = new DesiredCapabilities();
    String gridURL = "http://172.27.128.1:4444";
    switch (browser) {
      case "firefox":
        System.setProperty("webdriver.gecko.driver", "geckodriver.exe");
        return driver = new FirefoxDriver();
      case "safari":
        return driver = new SafariDriver();
      case "grid-safari":
        capabilities.setCapability("browserName", "safari");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      case "grid-firefox":
        capabilities.setCapability("browserName", "firefox");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      case "grid-chrome":
        capabilities.setCapability("browserName", "chrome");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      default:
       return driver = new ChromeDriver();
    }
  }

```

**Step 4**  Write a method to initialize the driver, for example
 `driver = pickBrowser("grid-firefox");`
![image](https://user-images.githubusercontent.com/104761482/201595745-944c7c8b-80f2-437e-b98f-197528282c86.png)


**1. To start the selenium server:**

For example java `-jar selenium-server-4.6.0.jar standalone`
![image](https://user-images.githubusercontent.com/104761482/201595042-9ea8dedd-db56-42bf-a438-ff7c965bf672.png))

2. Add an URL to your project, for example

![image](https://user-images.githubusercontent.com/104761482/201595216-5d07097f-eb45-4529-bc2e-f40c3c3b80ea.png)

`String gridURL = "http://172.21.160.1:4444/";`

![image](https://user-images.githubusercontent.com/104761482/201294845-e5f2b224-e97b-4dae-b98c-fd7928cd4e2e.png)

![image](https://user-images.githubusercontent.com/104761482/201294920-0f93d1b0-7c18-4c04-bd5d-b8193cb8ebbb.png)


For example:

```
public class BaseTest {

  WebDriver driver;
  String url;

  @BeforeSuite
  public static void chromeConfigs() {

    if (System.getProperty("os.name").toLowerCase().contains("win")) {
      System.setProperty("webdriver.chrome.driver", "chromedriver.exe");
    }
  }

  @Contract(value = " -> new", pure = true)
  @DataProvider(name = "invalidCredentials")
  public static Object[] @NotNull [] getCredentials() {

    return new Object[][]{
        {"invalid@class.com", "invalidPass"},
        {"d@class.com", ""},
        {"", ""}
    };
  }

  @BeforeClass
  @Parameters({"BaseURL"})
  public void launchBrowser(String BaseURL) throws MalformedURLException {
    driver = pickBrowser("grid-chrome");
    driver.manage().window().maximize();
    driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
    url = BaseURL;
    driver.get(url);
  }


  public WebDriver pickBrowser(String browser) throws MalformedURLException {
    DesiredCapabilities capabilities = new DesiredCapabilities();
    String gridURL = "http://172.27.128.1:4444";
    switch (browser) {
      case "firefox":
        System.setProperty("webdriver.gecko.driver", "geckodriver.exe");
        return driver = new FirefoxDriver();
      case "safari":
        return driver = new SafariDriver();
      case "grid-safari":
        capabilities.setCapability("browserName", "safari");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      case "grid-firefox":
        capabilities.setCapability("browserName", "firefox");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      case "grid-chrome":
        capabilities.setCapability("browserName", "chrome");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      default:
        return driver = new ChromeDriver();

    }
  }

  @AfterMethod
  public void tearDownBrowser() {
    driver.quit();
  }
}
```



Hope this at least helps you know where you stand.